packages/service-worker-mock/models/Client.js

```diff
@@ -2,14 +2,19 @@ const generateRandomId = require('../utils/generateRandomId');
 
 // https://developer.mozilla.org/en-US/docs/Web/API/Client
 class Client {
-  constructor(url, frameType) {
+  constructor(url, options) {
```

Comment:
It looks like passing in a `postMessageCallback` is not part of the Client spec, and causes a backwards incompatibility if someone is passing in frameType currently. Can you provide more details of the reasoning for this?

Reply:
I've passed this callback in order to pass a mocked function to properly test if post-message was called. I can pass is to a different parameter.

Reply:
Cool. Can you share a snippet of what your test looks like? We might be able to find a better way than adding non-spec code to the mocks 🤞🏻

Reply:
This is the test, it will be called when we intercept the fetch.

Reply:
```javascript
'use strict';

const updateClients = async (event, response) => {
    const clients = await self.clients.matchAll();
    const data = await response.json();
    const { clientId } = event;
    clients.forEach(client => {
        if (clientId !== client.id) {
            client.postMessage({
                data
            });
        }
    });
};

const cachedResourceOrFetch = async event => {
    // some more logic
    const response = fetch('url');
    updateClients(event, response.clone());
    return response.clone();
};

self.addEventListener('fetch', async event => {
    event.respondWith(cachedResourceOrFetch(event));
});
```

Reply:
I see. It's hard to suggest a better way to test it without the full example, but I think the simplest way would be to override the mock on the client itself. That way we're not adding non-spec code to the models themselves.
```
const client = await clients.openWindow(
    'http://localhost:3000',
    clientPostCallback
);
client. postMessage = jest.fn();
...
```

Reply:
^ Does that work for you? This should make it easy to verify that postMessage is being called (but that's not really a great unit to test).

I'm trying to have the mocks act exactly like the spec, and adding extra parameters is undesirable. There's no good way to update the postMessage method itself since that talks to a separate DOM environment. The only better way would be to use puppeteer and write a unit test to verify that the postMessage is working correctly in your app.